centreon/src/Core/AgentConfiguration/Application/UseCase/UpdateAgentConfiguration/UpdateAgentConfiguration.php

```diff
@@ -0,0 +1,168 @@
+ *
+ * For more information : contact@centreon.com
+ *
+ */
+
+declare(strict_types=1);
+
+namespace Core\AgentConfiguration\Application\UseCase\UpdateAgentConfiguration;
+
+use Assert\AssertionFailedException;
+use Centreon\Domain\Contact\Contact;
+use Centreon\Domain\Contact\Interfaces\ContactInterface;
+use Centreon\Domain\Log\LoggerTrait;
+use Centreon\Domain\Repository\Interfaces\DataStorageEngineInterface;
+use Core\AgentConfiguration\Application\Exception\AgentConfigurationException;
+use Core\AgentConfiguration\Application\Factory\AgentConfigurationFactory;
+use Core\AgentConfiguration\Application\Repository\ReadAgentConfigurationRepositoryInterface;
+use Core\AgentConfiguration\Application\Repository\WriteAgentConfigurationRepositoryInterface;
+use Core\AgentConfiguration\Domain\Model\AgentConfiguration;
+use Core\AgentConfiguration\Domain\Model\Poller;
+use Core\Application\Common\UseCase\ErrorResponse;
+use Core\Application\Common\UseCase\ForbiddenResponse;
+use Core\Application\Common\UseCase\InvalidArgumentResponse;
+use Core\Application\Common\UseCase\NoContentResponse;
+use Core\Application\Common\UseCase\NotFoundResponse;
+use Core\Application\Common\UseCase\PresenterInterface;
+use Core\MonitoringServer\Application\Repository\ReadMonitoringServerRepositoryInterface;
+use Core\Security\AccessGroup\Application\Repository\ReadAccessGroupRepositoryInterface;
+
+final class UpdateAgentConfiguration
+{
+    use LoggerTrait;
+
+    public function __construct(
+        private readonly ReadAgentConfigurationRepositoryInterface $readAcRepository,
+        private readonly WriteAgentConfigurationRepositoryInterface $writeAcRepository,
+        private readonly ReadAccessGroupRepositoryInterface $readAccessGroupRepository,
+        private readonly ReadMonitoringServerRepositoryInterface $readMonitoringServerRepository,
+        private readonly Validator $validator,
+        private readonly DataStorageEngineInterface $dataStorageEngine,
+        private readonly ContactInterface $user,
+    ) {
+    }
+
+    public function __invoke(
+        UpdateAgentConfigurationRequest $request,
+        PresenterInterface $presenter
+    ): void {
+        try {
+            if (! $this->user->hasTopologyRole(Contact::ROLE_CONFIGURATION_POLLERS_AGENT_CONFIGURATIONS_RW)) {
+                $this->error(
+                    "User doesn't have sufficient rights to access agent configurations",
+                    ['user_id' => $this->user->getId()]
+                );
+                $presenter->setResponseStatus(
+                    new ForbiddenResponse(AgentConfigurationException::accessNotAllowed())
+                );
+
+                return;
+            }
+
+            if (null === $agentConfiguration = $this->getAgentConfiguration($request->id)) {
+                $presenter->setResponseStatus(
+                    new NotFoundResponse('Agent Configuration')
+                );
+
+                return;
+            }
```

Comment:
This could be remain here, but I think we should considering for futur implementation to also move this logic into validators, as if the provided Id is not "valid" then we don't want to continue

Reply:
I get your logic but that would mean that the validation class would need to return the manipulated object and I don't like that